### PR TITLE
[es_PE] Add common first and last names

### PR DESCRIPTION
**Pull request by Oxicode**
### What is the reason for this PR?

- [x] A new feature
- [ ] Fixed an issue (resolve #ID)

### Author's checklist

- [x] Follow the [Contribution Guide](https://github.com/FakerPHP/Faker/blob/main/.github/CONTRIBUTING.md)
- [ ] New features and changes are [documented](https://github.com/FakerPHP/fakerphp.github.io)

### Summary of changes

Add names and Last Name common at Perú

### Review checklist

- [ ] All checks have passed
- [ ] Changes are approved by maintainer

### CODE REVIEW

**Comment by pimjansen** [comment]
There are a lot of changes in this PR that do not belong here.

Please review

File: src/Faker/Provider/es_PE/Company.php [hunk 1]
```diff
@@ -65,4 +65,24 @@ public function bs()
...
+     * 8 digits.
+     *
+     * @example '10164090581'
+     * @example '20105895410'
+     *
+     * @param bool $isNaturalPerson If set to false or empty, RUC starts with 20 and otherwise with 10
+     *
+     * @see http://www2.sunat.gob.pe/pdt/pdtModulos/independientes/p695/TipoDoc.htm
+     */
+    public static function ruc(bool $isNaturalPerson = false): string
```
Comment (bramceulemans) [inline src/Faker/Provider/es_PE/Company.php:82]: Remove unrelated changes.

File: test/Faker/Provider/es_PE/CompanyTest.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace Faker\Test\Provider\es_PE;
+
+use Faker\Provider\es_PE\Company;
+use Faker\Test\TestCase;
+
+/**
+ * @group legacy
+ */
+final class CompanyTest extends TestCase
```
Comment (bramceulemans) [inline test/Faker/Provider/es_PE/CompanyTest.php:11]: Remove unrelated changes.

**Review by bramceulemans — CHANGES_REQUESTED** [review]
I left some comments, please also provide a clear description together with a source.

**Comment by Oxicode** [comment]
Hola
Son commits acumulativos. Cuando mi otro PR sea fusionado este cambio será transparente.

**Comment by bramceulemans** [comment]
Translation: 
> They are cumulative commits.  When my other PR is merged this change will be transparent.

That doesn't change the fact that this PR includes unrelated changes, please apply this to a separate branch.

**Comment by Oxicode** [comment]
As I was saying, the changes you mention (RUC) when my other PR is merged are not tracked. It is the advantage of using GIT.
@bramceulemans, I use Google translate to communicate with you (because my english is very bad), how bad would it be if it were the other way around?

**Comment by pimjansen** [comment]
> As I was saying, the changes you mention (RUC) when my other PR is merged are not tracked. It is the advantage of using GIT.
> @bramceulemans, I use Google translate to communicate with you (because my english is very bad), how bad would it be if it were the other way around?

Please use english in communication for the project. 

Concerning the changes, you are partially right but the other PR is not merge and history can be rewritting which causes issues. Please ensure that your change contains code for this PR only and nothing else.

**Comment by Oxicode** [comment]
done. check again please.

File: src/Faker/Provider/es_PE/Person.php [hunk 3]
```diff
@@ -25,9 +25,9 @@ class Person extends \Faker\Provider\Person
     ];
 
     protected static $firstNameMale = [
-        'Aarón', 'Adrián', 'Agustín', 'Alan', 'Alejandro', 'Alex', 'Alexander', 'Alonso', 'Andrés', 'Anthony', 'Antonio', 'Axel', 'Benjamín',
```
Comment (pimjansen) [inline src/Faker/Provider/es_PE/Person.php:28]: Like other providers can we list each first letter on a new line for these lists here?

This will help in tracking changes.

I also miss a @see reference as mentioned in the guide. Where is the data coming from?

**Review by pimjansen — CHANGES_REQUESTED** [review]

File: src/Faker/Provider/es_PE/Person.php (hunk 3 above)
Comment (localheinz) [reply src/Faker/Provider/es_PE/Person.php:28]: Quick heads-up, @fzaninotto would usually reject pull requests modifying data sets, because they could break tests down-stream.

Here is an example: https://github.com/fzaninotto/Faker/pull/1775#issuecomment-525027301.

I have no objection to modifying data sets.

Comment (bramceulemans) [reply src/Faker/Provider/es_PE/Person.php:28]: I agree with the fact that changing datasets is fine.

Comment (pimjansen) [reply src/Faker/Provider/es_PE/Person.php:28]: Same here but would like to see a source instead of random names that someone likes.

**Comment by Oxicode** [comment]
ping @pimjansen  @bramceulemans

**Review by pimjansen — APPROVED** [review]

**Comment by bramceulemans** [comment]
Do you have a source for the names you added?

**Comment by Oxicode** [comment]
Mmm, I have put the names of my friends and family who live in Peru forever.
But here are some references:
image

https://diariocorreo.pe/peru/reniec-los-40-nombres-y-apellidos-mas-usuales-en-peru-833163/

**Comment by pimjansen** [comment]
@Oxicode please add a relevant source as mentioned in the contributions guide. Just a list of friends a families will not do. It should be a public and open licensed source.

If not available we need to decline this PR.

File: src/Faker/Provider/es_PE/Person.php [hunk 4]
```diff
@@ -25,49 +25,69 @@ class Person extends \Faker\Provider\Person
...
+        'Oliva', 'Olivia', 'Ornela',
+        'Paula', 'Paulina', 'Pilar',
+        'Rafaela', 'Rebeca', 'Regina', 'Renata', 'Romina',
+        'Salomé', 'Samantha', 'Sara', 'Silvana', 'Sofía', 'Sophie',  'Sara Sofía',
+        'Valentina', 'Valeria', 'Valery', 'Victoria', 'Violeta',
+        'Ximena',
+        'Zoe', 'Zoila',
     ];
 
     protected static $lastName = [
```
Comment (bramceulemans) [inline src/Faker/Provider/es_PE/Person.php:71]:
```suggestion
    /**
     * @see https://forebears.io/peru/surnames
     */
    protected static $lastName = [
```

**Comment by Oxicode** [comment]
> @Oxicode please add a relevant source as mentioned in the contributions guide. Just a list of friends a families will not do. It should be a public and open licensed source.
> 
> If not available we need to decline this PR.

Ridículo. En fin, espero que logres una buena administración del repo, desde el inicio quitaste todas las ganas de colaborar.

**Comment by bramceulemans** [comment]
Sorry that you feel this way. It is important to document sources for your datasets to be able to update them in the future. We also support a lot of languages, so a linked source is a good way for us to verify the data.

Translation:

> Lamento que te sientas así. Es importante documentar las fuentes de sus conjuntos de datos para poder actualizarlos en el futuro. También admitimos muchos idiomas, por lo que una fuente vinculada es una buena forma de verificar los datos.
